### PR TITLE
Generate *.changes file from git log in pre_checkin.sh script

### PR DESCRIPTION
**Pull request by davidcassany**
With this commit a *.changes file is generated using the git-log tool
formatted with the same template open build service tracks changes.

### CODE REVIEW

**Comment by davidcassany** [comment]
This PR is related to the open build service requests 617921~617940 and it should not be merged until those requests are accepted (as the *.changes file has been renamed and the script already uses the new name)

**Comment by davidcassany** [comment]
[Make sure changes file for images reflects the changes in container-images repo as it is expected](https://trello.com/c/ePnpLKIs/1035-make-sure-changes-file-for-images-reflects-the-changes-in-container-images-repo-as-it-is-expected)

File: pre_checkin.sh [hunk 1]
```diff
@@ -1,8 +1,31 @@
-#!/bin/bash 
+#!/bin/bash
 
-for product in kubic caasp; do
-    for file in *kiwi.ini; do
-        newfile=$(echo "${file}" | sed "s/^\(.*\).kiwi.ini/${product}-\1.kiwi/")
+## pre_ceckin.sh script is expected to be called from
```
Comment (Vogtinator) [inline pre_checkin.sh]: typo

File: pre_checkin.sh [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+## a local copy of the build service image after running
+## the _services file, e.g. `osc service disabledrun`
+
+dashes_line="-------------------------------------------------------------------"
+changes_head="${dashes_line}%n%cd - %ae%n"
+changes_body="%w(67)- %s %n%n%w(67,2,2)%b"
+date_format="%a %b %d %T %Z %Y"
```
Comment (Vogtinator) [inline pre_checkin.sh]: Dates should be in UTC, otherwise certain parsers get confused.

File: pre_checkin.sh [hunk 3]
```diff
@@ -1,8 +1,31 @@
...
+    changes_file="${image}.changes"
+
+    # *.changes file update from git-log
+    cp "${changes_file}" "${changes_file}.tmp"
+    pushd container-images 1> /dev/null || exit 1
+        git log --pretty=format:"${changes_head}%n${changes_body}" \
+            --date=format-local:"${date_format}" --no-merges --date-order \
+            --after="${last_changes_date}" -- pre_checkin.sh "${image}" \
+            | cat - "../${changes_file}.tmp" > "../${changes_file}" 
+    popd 1> /dev/null || exit 1
```
Comment (Vogtinator) [inline pre_checkin.sh]: set -e instead of "|| exit 1"?

**Comment by MaximilianMeister** [comment]
would be nice if we have one way of doing the changelog, as i think we're doing the same thing for the other packages in https://gitlab.suse.de/docker/automation/blob/master/concourse/package.sh#L21-46

**Comment by davidcassany** [comment]
@Vogtinator Thanks for the review, should be addressed now.

@MaximilianMeister Right, using the same format now.

**Pull request merged**